packages/scaffold-config/src/supportFile.ts

```diff
@@ -0,0 +1,91 @@
+import type { CodeLanguage } from '@packages/types'
+import dedent from 'dedent'
+import type { WizardFrontendFramework } from '.'
+
+export function supportFileE2E (language: CodeLanguage['type']) {
+  return dedent`
+    // ***********************************************************
+    // This example support/e2e.${language} is processed and
+    // loaded automatically before your test files.
+    //
+    // This is a great place to put global configuration and
+    // behavior that modifies Cypress.
+    //
+    // You can change the location of this file or turn off
+    // automatically serving support files with the
+    // 'supportFile' configuration option.
+    //
+    // You can read more here:
+    // https://on.cypress.io/configuration
+    // ***********************************************************
+
+    // Import commands.js using ES2015 syntax:
+    import './commands'
+
+    // Alternatively you can use CommonJS syntax:
+    // require('./commands')
+  `
+}
+
+export function supportFileComponent (language: CodeLanguage['type'], framework: WizardFrontendFramework) {
+  const supportFileTemplate = dedent`
+    // ***********************************************************
+    // This example support/component.${language} is processed and
+    // loaded automatically before your test files.
+    //
+    // This is a great place to put global configuration and
+    // behavior that modifies Cypress.
+    //
+    // You can change the location of this file or turn off
+    // automatically serving support files with the
+    // 'supportFile' configuration option.
+    //
+    // You can read more here:
+    // https://on.cypress.io/configuration
+    // ***********************************************************
+
+    // Import commands.js using ES2015 syntax:
+    import './commands'
+
+    // Alternatively you can use CommonJS syntax:
+    // require('./commands')
+  `
+
+  const exampleUse = dedent`
+    // Example use:
+    // cy.mount(${framework.mountModule === 'cypress/react' ? '<MyComponent />' : 'MyComponent'})
+  `
+
+  const NEWLINE = '\n\n'
+
+  if (language === 'ts') {
+    const registerMount = dedent`
+      import { mount } from '${framework.mountModule}'
+      import type { MountReturn } from '${framework.mountModule}'
+
+      // Augment the Cypress namespace to include type definitions for
+      // your custom command.
+      // Alternatively, can be defined in cypress/support/component.d.ts
+      // with a <reference path="./component" /> at the top of your spec.
+      declare global {
+        namespace Cypress {
+          interface Chainable {
+            mount(...): Cypress.Chainable<MountReturn>
```

Comment:
This `...` here isn't meant to be what we actually ship, I mostly meant it to indicate that it should be is supposed to be filled in with the appropriate mount signature dependent on the framework chosen

Reply:
I guess it could be as simple as 
```ts
declare global {
  namespace Cypress {
    interface Chainable {
      mount: typeof mount
    }
  }
}      
```

Reply:
:man_facepalming:

Reply:
Done. https://github.com/cypress-io/cypress/pull/20933/commits/d304fba970835a9a96cb5af8a7e7ce18d4cc7db0